### PR TITLE
test: spec out new structure

### PR DESCRIPTION
**Pull request by l-zeuch**
Draft PR to discuss and spec out a new test structure.

A new draft is provided alongside, so that we have some starting point in (re)-structuring our tests, but its definitely up to discussion and changes.

Once we're happy with a specific aspect and have changed it around a little, a respective commit will follow.

I'd suggest we make use of review comments, as they can be attached to specific lines within the patch(es) and GitHub will take care of having them attached to the correct line.

**Terms**
- [x] I agree to follow this project's [Code of Conduct](CODE_OF_CONDUCT.md)
- [x] I have read and understood this project's [Contributing Guidelines](../CONTRIBUTING.md)

### CODE REVIEW

**Review by LRitzdorf — COMMENTED** [review]
This looks good so far! I do like this new structure.

Also, I haven't used review comments before, but that does sound like a useful feature.

File: test/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+## Adding New Tests
+
+This project uses a framework to test its Vim Script: [vader.vim](https://github.com/junegunn/vader.vim). Before adding new
+tests, it's definitely recommended to read its documentation.
+
+When adding new tests, only one specific feature should be tested within one test file. This is to avoid hard-to-read,
+hard-to-debug test files as well pipeline output. It furthermore makes it simpler to expand already existing tests.
+
+Afterwards, `Include` your newly added test file in `test.vader` and make sure all tests are passing.
```
Comment (l-zeuch) [inline test/README.md:23]: One thing that popped up regarding `Include`:

With the old structure, each individual file was one test suite. Now, that we *plan to* include files, that collapses down to one big test suite.

Let me know what you think -- personally, I prefer the more detailed logging that we currently have, as it makes debugging easier. However, I can also see why one would prefer one large suite.

Screenshots of the two logs attached below.

Logs with `Include` directive

Individual suites (w/ new directory structure)

Comment (LRitzdorf) [reply test/README.md:23]: I think I agree, actually — having multiple suites is nicer in terms of granularity.

Comment (l-zeuch) [reply test/README.md:23]: Will push the appropriate changes 👍

**Comment by l-zeuch** [comment]
Progress is kind of stalling, as I am not sure what we have to test for and what is technically possible to test for (within reason).

I'll have to read through our project and figure things out as we go, however advice is greatly appreciated.

**Review by LRitzdorf — APPROVED** [review]
lgtm! And the Neovim tests are even passing again :shipit:

**Comment by l-zeuch** [comment]
:shipit:

**Pull request merged**